src/mbedtls_context.cpp

```diff
@@ -746,7 +748,7 @@ namespace sockpp {
                 *flags |= MBEDTLS_X509_BADCERT_OTHER;
             }
         } else {
-            if (pinned_cert_) {
+            if (pinned_cert_ || callback) {
```

Comment:
Don't we need to check the result of callback?

Reply:
No, we don't need to because we only care about the verification result of the leaf cert.